makeelf/elf.py

```diff
@@ -379,11 +384,11 @@ def append_segment(self, sec_id, addr=None, mem_size=-1, flags='rwx'):
         return self._append_segment(ptype=PT.PT_LOAD, vaddr=addr, paddr=0,
                 file_size=Shdr.sh_size, mem_size=mem_size, flags=p_flags)
 
-    def _append_segment(self, ptype, vaddr, paddr, file_size, mem_size, flags=0):
+    def _append_segment(self, ptype, vaddr, paddr, file_size, mem_size, flags=0, sections=[]):
```

Comment:
I would suggest initializing this with `None` and construct the empty list in the function, as mutable defaults are problematic: https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

Just hit a but myself when I was trying to append to segment's sections...

Reply:
Same with `Elf32_Phdr` constructor

Reply:
Ooo, I learned something today! :star_struck: Thanks, I'll fix that.